docs/web_reference.rst

```diff
@@ -1130,38 +1130,61 @@ duplicated like one using :meth:`Application.copy`.
-          of :attr:`Application.debug`.
-          The :class:`Application`'s debug mode setting should be used
-          as a single point to setup a debug mode.
-
-      You should pass result of the method as *protocol_factory* to
-      :meth:`~asyncio.AbstractEventLoop.create_server`, e.g.::
-
-         loop = asyncio.get_event_loop()
-
-         app = Application(loop=loop)
-
-         # setup route table
-         # app.router.add_route(...)
-
-         await loop.create_server(app.make_handler(),
-                                  '0.0.0.0', 8080)
-
-      ``secure_proxy_ssl_header`` keyword parameter
-      can be used to detect request scheme::
-
-         await loop.create_server(app.make_handler(
-            secure_proxy_ssl_header='X-Forwarded-Proto'),
-            '0.0.0.0', 8080)
+    Creates HTTP protocol factory for handling requests.
+
+    :param handler: Request handler. Default:
+      :class:`aiohttp.web.RequestHandler`.
+    :param str secure_proxy_ssl_header: Secure proxy SSL header. Default:
+      `None`.
+    :param int keepalive_timeout: Number of seconds before closing Keep-Alive
+        connection. Default: ``75`` seconds (NGINX's default value).
+    :param bool tcp_keepalive: Enable TCP Keep-Alive. Default: ``True``.
+    :param slow_request_timeout: Slow request timeout. Default: ``0``.
+    :param logger: Custom logger object. Default:
+        :data:`aiohttp.log.server_logger`.
+    :param access_log: Custom logging object. Default:
+        :data:`aiohttp.log.access_logger`.
+    :param str access_log_format: Access log format string. Default:
+        :attr:`helpers.AccessLogger.LOG_FORMAT`.
+    :param bool debug: Switches debug mode. Default: ``False``.
+
+      .. deprecated:: 1.0
+
+        The usage of ``debug`` parameter in :meth:`Application.make_handler`
+        is deprecated in favor of :attr:`Application.debug`.
+        The :class:`Application`'s debug mode setting should be used
+        as a single point to setup a debug mode.
+
+    :param int max_line_size: Optional maximum header line size. Default:
+      ``8190``.
+    :param int max_headers: Optional maximum header size. Default: ``32768``.
+    :param int max_field_size: Optional maximum header field size. Default:
+      ``8190``.
+    :param eof_exc_class: Parameter of :class:`aiohttp.parsers.StreamParser`.
+      Default: `RuntimeError <https://docs.python.org/3/library/exceptions.html#RuntimeError>`_.
+    :param buf: Parameter of :class:`aiohttp.parsers.StreamParser`.
+      Default: ``None``.
+
+
+    You should pass result of the method as *protocol_factory* to
+    :meth:`~asyncio.AbstractEventLoop.create_server`, e.g.::
+
+       loop = asyncio.get_event_loop()
+
+       app = Application(loop=loop)
+
+       # setup route table
+       # app.router.add_route(...)
+
+       await loop.create_server(app.make_handler(),
+                                '0.0.0.0', 8080)
+
+    ``secure_proxy_ssl_header`` keyword parameter
```

Comment:
Move it into `:param secure_proxy_ssl_header:` section.
Drop the code example, it's just wrong.
The value should be a tuple `secure_proxy_ssl_header=('X-Forwarded-Proto', 'https')` but I'm considering to deprecate it but support `X-Forwarded-Proto` implicitly in `web.Request.scheme`.

See also #1134